02-peer-protocol.md

```diff
@@ -50,10 +67,451 @@ pubkey corresponding to the funding output nothing prevents duplicative channel
+  - MUST NOT re-transmit inputs it has already received from the peer
+  - MUST send a non-malleable (segwit) UTXO.
+  - if is the `initiator`:
+    - MUST send even `serial_id`s
+  - if is the `contributor`:
+    - MAY omit this message
+    - MUST send odd `serial_id`s
+
+The receiving node:
+  - MUST add all received inputs to the funding transaction
+  - MUST fail the transaction collaboration if:
+    - it receives a duplicate input to one it sent previously
+    - it receives an input that would create a malleable transaction id (e.g. pre-Segwit)
+    - it receives an unconfirmed input
+    - it receives a duplicate `serial_id`
+    - it receives a `serial_id` from the peer with the incorrect parity
+
+#### Rationale
+Each node must have a complete set of the transaction inputs
+to derive the funding transaction and subsequent commitment signatures.
+
+`serial_id` is a randomly chosen number which uniquely identifies this input.
+Inputs in the final transaction will be sorted by `serial_id`.
+
+`prevtx_tx` is the serialized transaction whose output this node is spending. Must be sent to verify that output is in fact a SegWit output.
+
+ `prevtx_vout` output index of the output being spent.
+
+`max_witness_len` is the total serialized length of the
+witness data that will be supplied (e.g. sizeof(varint) +
+sizeof(witness) for each) in `funding_signed2`.
+
+`input_info`.`script` is the script signature field for the input.
+Only applicable for P2SH-wrapped inputs. The length of the
+script should not be included in the transmitted `script` data.
+
+Native SegWit inputs (P2WPKH and P2WSH) inputs will have an
+empty `script` field See [BIP141](https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki#examples).
+
+#### Proof of Discrete Log Equivalence
+
+The `podle_proof` TLV type furnishes the data necessary to verify
+a PoDLE commitment, as exchanged during the initiation phase.
+
+`e` is the sha256(kG || kJ || P || P2 || `node_id`) where:
+  - `kG` is the compressed point representation of a randomly chosen 32-byte nonce
+    , `k, multiplied by the generator point `G`
+  - `kJ` is the compressed point representation of `k` by the NUMS point `J`
+  - `P` is the compressed point representation of the utxo secret key (`x`)
+     times the generator point, `xG`, or `pubkey` for secret `x`
+  - `P2` is the compressed point representation of the point `J` by `x`, or `xJ`
+  - `node_id` is the 33-byte id of the node receiving this proof
+
+`sig` is the signature, `k + x * e` (over the secp256k1 finite field)
+
+Note that `x` is the private key of a pubkey that the utxo is locked to.
+
+The commitment, `podle_h2`, is the sha256 hash of the compressed point
+representation of `P2`, or `xJ`.
+
+The signature can be verified by the recipient as follows:
+  - `kG = sG - eP`, where `s` is the received `sig` value
+  - `kJ = sJ - eP2`
+  - `sha256(kG || kJ || P || P2 || node_id)` where the point value is
```

Comment:
is this meant to be checked against `e`?

Reply:
i don't understand the question.

Reply:
I might be misunderstanding but is this last step of verification to check that `e` is equal to this hash?